packages/frontend/app/routes/payments.incoming.$incomingPaymentId.tsx

```diff
@@ -0,0 +1,188 @@
+import type { LoaderArgs } from '@remix-run/node'
+import { json } from '@remix-run/node'
+import { Outlet, useLoaderData } from '@remix-run/react'
+import { useState } from 'react'
+import { z } from 'zod'
+import { Badge, PageHeader } from '~/components'
+import { Button } from '~/components/ui'
+import { getIncomingPayment } from '~/lib/api/payments.server'
+import { badgeColorByState, formatAmount, prettify } from '~/shared/utils'
+
+export async function loader({ params }: LoaderArgs) {
+  const incomingPaymentId = params.incomingPaymentId
+
+  const result = z.string().uuid().safeParse(incomingPaymentId)
+  if (!result.success) {
+    throw json(null, {
+      status: 400,
+      statusText: 'Invalid incoming payment ID.'
+    })
+  }
+
+  const incomingPayment = await getIncomingPayment({ id: result.data })
+
+  if (!incomingPayment) {
+    throw json(null, { status: 400, statusText: 'Incoming payment not found.' })
+  }
+
+  return json({
+    incomingPayment: {
+      ...incomingPayment,
+      createdAt: new Date(incomingPayment.createdAt).toLocaleString(),
+      expiresAt: new Date(incomingPayment.expiresAt).toLocaleString()
+    }
+  })
+}
+
+export default function ViewIncomingPaymentPage() {
+  const { incomingPayment } = useLoaderData<typeof loader>()
+  const [showMetadata, setShowMetadata] = useState(false)
+
+  return (
+    <div className='pt-4 flex flex-col space-y-4'>
+      <div className='flex flex-col rounded-md bg-offwhite px-6'>
+        {/* Incoming Payment General Info */}
+        <PageHeader className='!justify-end'>
+          <Button aria-label='go back to payments page' to='/payments'>
+            Go to payments page
+          </Button>
+        </PageHeader>
+        <div className='grid grid-cols-1 py-3 gap-6 md:grid-cols-3 border-b border-pearl'>
+          {/* Incoming Payment General Info*/}
+          <div className='col-span-1 pt-3'>
+            <h3 className='text-lg font-medium'>General Information</h3>
+            <p className='text-sm'>
+              Created at {incomingPayment.createdAt}{' '}
+              <Badge color={badgeColorByState[incomingPayment.state]}>
+                {incomingPayment.state}
+              </Badge>
```

Comment:
I feel like the state should also be shown along side the other information (on the right column).

Reply:
I added it to the group of fields as a badge. I've gone back and forth on this. I also considered just formatting the plain text value (ie: "Comepleted") and welcome opinions if anyone has any, although this is probably a minor point.

I'm not sure I like the badge as a field value but I do like the badge. I think the "State" title is a bit redundant - the badge itself indicates its some sort of state. It also breaks the uniformity of the field values a bit. This is what motivated me to move it outside the group of fields (although I wasn't entirely sold on the location I picked). A formatted text value is OK but I like the badge because the color immediately highlights the general state (green: good, red: bad, etc). The state is also displayed as a badge in the payments page, so using a badge would be more consistent.

![image](https://github.com/interledger/rafiki/assets/12960453/01d16bee-da93-4eab-9c69-13cc0a59ab76)

![image](https://github.com/interledger/rafiki/assets/12960453/b080c966-7719-46e4-931a-dfd7a24042ce)